src/Illuminate/Support/Str.php

```diff
@@ -2077,6 +2077,30 @@ public static function freezeUlids(?Closure $callback = null)
         return $ulid;
     }
 
+    /**
+     * Clip a string in the middle if it exceeds the given length.
+     *
+     * For example, "SuperLongFileName.ext" with a max of 15 would become
+     * "SuperL…ame.ext".
+     *
+     * This is useful for displaying long identifiers (filenames, hashes, etc.)
+     * while keeping both the beginning and end visible.
+     *
+     * @param  string  $value
+     * @param  int  $max
```

Comment:
You can narrow this further down:
```suggestion
     * @param  non-negative-int  $max
```